src/Compilers/CSharp/Portable/Binder/InMethodBinder.cs

```diff
@@ -274,13 +274,6 @@ internal static bool ReportConflictWithParameter(Symbol parameter, Symbol newSym
                     return false;
                 }
 
-                if (newSymbolKind == SymbolKind.Parameter || newSymbolKind == SymbolKind.Local)
```

Comment:
Expand the review up a few lines and you'll see an identical check being made which unconditionally returns in the `if` body.  Hence this code path should never get hit.